csharp/src/SeedLang/Runtime/BuiltinsDefinition.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright 2021-2022 The SeedV Lab.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using SeedLang.Common;
+using SeedLang.Runtime.HeapObjects;
+
+namespace SeedLang.Runtime {
+  using Range = HeapObjects.Range;
+
+  // The static class to define all builtin functions.
+  internal static class BuiltinsDefinition {
+    public const string PrintVal = "__printval__";
+    public const string Abs = "abs";
+    public const string All = "all";
+    public const string Any = "any";
+    public const string Append = "append";
+    public const string Dir = "dir";
+    public const string Len = "len";
+    public const string List = "list";
+    public const string Print = "print";
+    public const string Range = "range";
+    public const string Round = "round";
+    public const string Slice = "slice";
+    public const string Sum = "sum";
+
+    public static Dictionary<string, VMValue> Variables = new Dictionary<string, VMValue> {
```

Comment:
I think this definition dictionary initialization process can be simplified by C#'s Attribute classes.

For example, considering defining module method function like below:

```
[ModuleFunc(name = PrintVal)]
private static VMValue PrintValFunc(ValueSpan args, INativeContext context) {
  ...
}
```

With the above attribute-decorated functions, the initializer can access the attribute info of every function then create the dictionary accordingly.

Cons of the above solution: it requires C# reflection to read attribute info. I am not sure if Unity's managed code stripping https://docs.unity3d.com/Manual/ManagedCodeStripping.html will remove the reflection logic somehow.

Reply:
Re-implement by reflection without attribute. We can change the implement later if Unity removes these logic after testing in Unity 3D environment.

Reply:
OK. This solution looks good for now.